server/src/main/java/org/candlepin/pinsetter/tasks/ImportJob.java

```diff
@@ -0,0 +1,135 @@
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.pinsetter.tasks;
+
+import static org.quartz.JobBuilder.newJob;
+
+import org.candlepin.common.exceptions.BadRequestException;
+import org.candlepin.common.exceptions.IseException;
+import org.candlepin.controller.ManifestManager;
+import org.candlepin.model.ImportRecord;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+import org.candlepin.pinsetter.core.model.JobStatus;
+import org.candlepin.sync.ConflictOverrides;
+import org.candlepin.sync.ImporterException;
+import org.candlepin.sync.SyncDataFormatException;
+import org.candlepin.util.Util;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+
+/**
+ * Runs an asynchronous manifest import. This job expects that the manifest file
+ * was already uploaded.
+ */
+public class ImportJob extends UniqueByEntityJob {
+
+    protected static final String STORED_FILE_ID = "stored_manifest_file_id";
+    protected static final String CONFLICT_OVERRIDES = "conflict_overrides";
+    protected static final String UPLOADED_FILE_NAME = "uploaded_file_name";
+
+    private static Logger log = LoggerFactory.getLogger(ImportJob.class);
+
+    private OwnerCurator ownerCurator;
+    private ManifestManager manifestManager;
+
+    @Inject
+    public ImportJob(OwnerCurator ownerCurator, ManifestManager manifestManager) {
+        this.ownerCurator = ownerCurator;
+        this.manifestManager = manifestManager;
+    }
+
+    @Override
+    public void toExecute(JobExecutionContext context) throws JobExecutionException {
+        JobDataMap map = context.getMergedJobDataMap();
+        String ownerKey = (String) map.get(JobStatus.TARGET_ID);
+        ConflictOverrides overrides = new ConflictOverrides((String[]) map.get(CONFLICT_OVERRIDES));
+        String storedFileId = (String) map.get(STORED_FILE_ID);
+        String uploadedFileName = (String) map.get(UPLOADED_FILE_NAME);
+
+        Throwable caught = null;
+        Owner targetOwner = null;
+        try {
+            targetOwner = ownerCurator.lookupByKey(ownerKey);
+            if (targetOwner == null) {
+                //FIXME This should be internationalized.
+                context.setResult("Nothing to do. Owner no longer exists.");
```

Comment:
> While this made sense for ConsumerComplianceJob ( no point in calculating compliance of a consumer that has been deleted ), is the same true for import as well? I feel if some one explicitly requested an import, and owner no longer exists, we should error out the job.. we do face the same problem with spec tests where owner is already cleaned up though, so im split. thoughts?

Old behavior was to throw a NotFoundException. Perhaps we should error the job as you've said. I'll make that change.